lib/govuk_publishing_components/presenters/public_layout_helper.rb

```diff
@@ -69,20 +100,43 @@ def footer_track_actions
         FOOTER_TRACK_ACTIONS
       end
 
-      def generate_data_attribute(link, track_action)
+      def generate_data_attribute(link, track_action, ga4_attributes)
+        if ga4_attributes
+          ga4_link = {
+            "event_name": "navigation",
+            "type": "footer",
+            "index": {
+              "index_link": ga4_attributes[:index_link].to_s,
+              "index_section": ga4_attributes[:index_section].to_s,
+              "index_section_count": "5",
```

Comment:
Is `index_section_count` always 5? We've got examples in the component guide where the footer doesn't have all the links, does this line need to call the counting method?

Reply:
@andysellick I think for GOV.UK it is. I guess I could do a counting method but that method will still involve hard coding so I think the only benefit that would give us is that if the Topics / Government activity was given an extra section then this value would auto change, but for the other sections we'd probably have to manually change it.

Reply:
In the past we have added more sections to the footer, which would change this number (for example there was a coronavirus section I think, and definitely a Brexit one). 

Rather than block this PR (because no change to these sections is planned in the immediate future) can you record this on the tech debt card as something to either come up with a test that would break or a bit of documentation we could insert somewhere to highlight to future developers that this should change if they add new sections?

Reply:
@andysellick Thanks, I've added it to the tech debt doc.